sdcm/nemesis.py

```diff
@@ -1688,6 +1691,40 @@ def disrupt_nodetool_enospc(self, sleep_time=30, all_nodes=False):
                     finally:
                         clean_enospc_on_node(target_node=node, sleep_time=sleep_time)
 
+    def disrupt_end_of_quota_nemesis(self, sleep_time=30):
+        """
+        Nemesis flow
+        ---------------------
+        1. Enable quota on the node.
+        2. Check disk usage.
+        3. Define quota size same as disk usage + 3GB.
+        4. Approach end of quota in a loop with fallocate file that takes 90% of quota size each iteration.
+        5. Wait for end of quota message to appear in the log - "Disk quota exceeded" / Except I/O Error if happens.
+        6. Remove the sparse files.
+        7. Remove the quota by setting it to 0.
+        8. Restart scylla server.
+        9. Verify scylla is up.
+        """
+
+        node = self.target_node
+        if self._is_it_on_kubernetes():
+            raise UnsupportedNemesis("Skipping nemesis for kubernetes")
+
+        result = node.remoter.run('cat /proc/mounts')
+        if '/var/lib/scylla' not in result.stdout:
+            raise UnsupportedNemesis("Scylla doesn't use an individual storage, skip end of quota test")
+
+        enable_quota_on_node(node)
+        quota_enabled = is_quota_enabled_on_node(node)
+        if not quota_enabled:
+            raise QuotaConfigurationFailure("Failed to configure quota on the cluster")
+
+        with ignore_disk_quota_exceeded_errors(node):
+            with configure_quota_on_node_for_scylla_user_context(node) as quota_size:
+                write_data_to_reach_end_of_quota(node, quota_size)
+            node.restart_scylla_server()
```

Comment:
this is happening too quickly, you should align to the flow done in `clean_enospc_on_node`
it's first give something like 30sec for the scylla to shut down on it's own,
clears the temporary file
wait more 15sec
and then restarts scylla

if you are doing stop/restart on you own before scylla shut itself down.

you are hitting the systemd timeout, and core dumps scylla.

please stick to the same exact flow of ENOSPC, scylla behavior needs to be identical (from scylla code POV)
you trying total different flow here, is kind of making all kind of assumptions which aren't exactly accurate

Reply:
Even longer than 30sec, we should make sure cluster react to the node failure.